src/python/convert_numpy_features_to_protos_test.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import numpy as np
+from pathlib import Path
+
+import convert_numpy_features_to_protos as converter
+
+
+def test_convert_to_protos():
+    features = np.array([[1, 2, 3], [4, 5, 6]])
+    feature_type = "Test"
+
+    protos = converter.convert_to_protos(features, feature_type)
+    np.testing.assert_equal(protos[0].values, features[0, :])
+    np.testing.assert_equal(protos[1].values, features[1, :])
+
+    assert protos[0].type == feature_type
+    assert protos[1].type == feature_type
+
+    assert protos[0].size == 3
+    assert protos[1].size == 3
+
+
+def test_convert_to_proto_not_valid():
+    features = np.zeros((2, 3, 4))
+    with pytest.raises(AssertionError):
+        protos = converter.convert_to_protos(features, "Test")
+
+
+def test_save_protos_to_files(tmp_path):
```

Comment:
Question out of curiosity, where does the `tmp_path` come from?

Reply:
I use `tmp_path` whevener I need a clear new directory where temporary file needed for test will be created.
It get's cleaned after every successful test. Whenever the test fails, there is a possibility to check that folder to see what actually happened.
I think this web page explains in more details how exactly this works in Python and how pytest uses it.
https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
This is awesome! I did not know about this!